BrainPortal/app/controllers/invitations_controller.rb

```diff
@@ -48,18 +48,21 @@ def new #:nodoc:
   # Send an invitation
   def create #:nodoc:
     @group          = Group.find(params[:group_id])
-    user_ids        = params[:user_ids].map(&:to_i)
+    user_ids        = (params[:user_ids] || []).map(&:to_i)
     already_sent_to = Invitation.where(sender_id: current_user.id, active: true, user_id: user_ids, group_id: @group.id).all.map(&:user_id)
     rejected_ids    = user_ids & already_sent_to
-    if rejected_ids.present?
+
+    if user_ids.empty?
+      flash_message = "\nYou should select at the least one user."
+    elsif rejected_ids.present?
       flash_message = "\n#{User.find(rejected_ids).map(&:login).join(", ")} already invited."
     end
 
     @users = User.find((user_ids - already_sent_to) & current_user.visible_users.map(&:id))
 
     unless @group.can_be_edited_by?(current_user) && @users.present?
       flash[:error]  = "Could not send the requested invitations."
-      flash[:error] += flash_message if rejected_ids.present?
+      flash[:error] += flash_message if defined? flash_message
```

Comment:
Don't use `defined?` it will return true even if the variable isn't set to a true value. Why not simply `if flash_message.present?`

Reply:
do not see problem if set to non-true value. But if to confusing or looks hackish can refactor with present, no problem.

Reply:
There IS a problem. It's a bug. defined?() returns the status of a variable as a string. A string is true. Try it yourself:

```
shell> cat x.rb 
if false
  mess = 'xyz'
elsif nil
  mess = 'xyz2'
end
puts "doing this" if defined? mess
shell> ruby x.rb 
doing this
```

Reply:
The condition for appending the flash_message is not whether or not it's defined, it's whether or not it's present. These are different. nil is defined, but not present. If you take this line just by itself with no other context you shoudl be able to see it can raise an exception:

```ruby
      flash[:error] += flash_message if defined? flash_message
```

when flash_message is nil, because you're going to try do string concatenation with nil.

Reply:
ok got it now

Reply:
BTW there is some dead code inside the method (the last branching) that I can get rid of

Reply:
Yes remove the dead code. Good catch.

Reply:
fixed